xbmc/Application.h

```diff
@@ -381,6 +381,50 @@ class CApplication : public CXBApplicationEx, public IPlayerCallback, public IMs
   */
   void UnlockFrameMoveGuard();
 
+  /*!
+  \brief Returns true if Application is currently playing an ISO stack
+  */
+  bool IsPlayingISOStack() const { return m_itemCurrentFile->IsStack() && m_currentStack->Size() > 0 && m_currentStackIsDiscImageStack; }
```

Comment:
those are a lot of new methods for CApplication. Does it make sense to factor this out in a separare class like PlaylistPlayer? Something like CStackPlayer?

Reply:
these are more helper methods to factor out functionality that was repeated over and over... I'm ok putting these in another class but it's really tightly intertwined with CApplication private members... Thoughts?

Reply:
> but it's really tightly intertwined with CApplication private members

I think we should take the chance and try to resolve this.